src/librustc_typeck/check/demand.rs

```diff
@@ -397,6 +398,29 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                         } else {
                             String::new()
                         };
+                        if let Some(hir::Node::Expr(hir::Expr {
```

Comment:
With this addition the function is ~200+ LOC -- can we avoid making it bigger? =P

Reply:
I'm focusing on slimming/splitting `Parser` now. I'll move to other parts of the compiler once in done with that.

Reply:
Cool, just don't forget about it... =)